proseco/guide.py

```diff
@@ -340,6 +349,45 @@ def get_initial_guide_candidates(self):
         return cand_guides
 
 
+
+def check_fid_trap(cand_stars, fids, dither):
+    """
+    Search for guide stars that would cause the fid trap issue and mark as spoilers.
+    """
+
+    spoilers = np.zeros(len(cand_stars)).astype(bool)
+    rej = []
+
+    if len(fids) == 0:
+        return spoilers, []
+
+    if not isinstance(dither, ACABox):
```

Comment:
At this point in the code `dither` must be an `ACABox`, right?  This function is not really called as a standalone thing and you can specify in the docstring that is has to an `ACABox`.